File: java/src/data_structures/LinkedList.java
package data_structures;

public class LinkedList {
    public Node head; // head
     public class Node{ //Inner class
        byte data;
        public Node next;
        Node (byte d){ //Node constructor
            data = d;
            next = null; }
    }

    //O(n) TRAVERSE
    public void traverse(){
         Node temp = head;
         do{
             System.out.print(temp.data + " -> ");
             temp = temp.next;
         }while(temp != null);
    }

    //O(1) PUSH
    public void push(byte new_data){
        Node new_node = new Node(new_data); //create new node with new data
        new_node.next = head; //make new node point to head
        head = new_node; //move head to new node
    }

    //O(n) INSERT
    public void insertAfter(Node prev_node, byte new_data){
         if (prev_node == null){ //check if previous node exists
             System.out.println("Previous node is null");
             return;
         }
         Node new_node = new Node(new_data); //create new node with new data
         new_node.next = prev_node.next; //set new node point to previous node
         prev_node.next = new_node; //set previous node point to new node
    }

    //O(n) APPEND
    public void append(byte new_data){
        Node new_node = new Node(new_data); //create new node with new data
        if (head == null){ //check if linkedList is empty
            head = new Node(new_data); //add new node to populate empty linkedList
            return;
        }
        new_node.next = null; //set new node point to null
        Node last = head; //find last node by traversing from the head
        while (last.next != null) last = last.next; //while last pointer is not null---traverse
        last.next = new_node; //change last pointer to new node;
    }
}

File: python/data_structures/linked_list.py
# create a node class
class Node:
    def __init__(self, data):
        self.data = data  # create data attribute to contain data
        self.next = None  # create pointer to point to next node


# create the linked list class
class SinglyLinkedList:
    def __init__(self):
        self.head = None  # create head attribute for all linked lists

    def push(self, new_data):
        new_node = Node(new_data)  # create the new node
        new_node.next = self.head  # make new node point to old head
        self.head = new_node  # make the new node new head

    def insertAfter(self, position, new_data):  # take position and new data
        new_node = Node(new_data)  # create the new node
        if self.head is None:
            self.head = new_node
            return
        temp = self.head  # set temp as head to start from the beginning of the list
        k = 0
        while k < position:  # loop through until the position has been reached
            k += 1
            temp = temp.next  # this will allow you to move up the list
        new_node.next = temp.next
        temp.next = new_node

    def append(self, new_data):
        new_node = Node(new_data)
        # if list is empty, set new_node as head
        if self.head is None:
            self.head = new_node
            return

        # else traverse till the last node
        last = self.head
        while last.next:
            last = last.next

        # change the next of last node
        last.next = new_node

    # remove elements, ADD CHECKERS FOR EMPTY LIST OR SOME OTHER CASES
    def remove(self, position):
        temp = self.head
        j = 0

        while j < position - 1 and temp.next is not None:  # while previous of target position is not reached
            temp = temp.next  # and next node is not None
            j += 1  # move temp forward
        temp.next = temp.next.next  # once index position has been reached point the pointer
        # to the pointer of the next node, this will cut the target index position from the list
        # idk what happens to it or how to properly remove it from memory

    def traverse(self):
        temp = self.head
        while temp:  # move forward from the head until it is not None and print the data
            print(str(temp.data) + " ->", end=' ')
            temp = temp.next